src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -906,7 +875,7 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
 
     } else {
       keyTypes match {
-        case Array(TVariant) =>
+        case Array(`vSignature`) =>
```

Comment:
I can't see the rest of the code easily but I think you need to fix the error message below